piptools/scripts/compile.py

```diff
@@ -33,11 +34,24 @@
 )
 from ..writer import OutputWriter
 
+ALL_BUILD_DISTRIBUTIONS = ("editable", "sdist", "wheel")
 DEFAULT_REQUIREMENTS_FILE = "requirements.in"
 DEFAULT_REQUIREMENTS_OUTPUT_FILE = "requirements.txt"
 METADATA_FILENAMES = frozenset({"setup.py", "setup.cfg", "pyproject.toml"})
 
 
+def _build_requirements(src_dir: str, distributions: Iterable[str]) -> set[str]:
+    builder = ProjectBuilder(src_dir, runner=pyproject_hooks.quiet_subprocess_runner)
+    # It is not clear that it should be possible to use `get_requires_for_build` with
+    # "editable" but it seems to work in practice.
+    return set(
+        itertools.chain(
```

Comment:
Use `itertools.chain.from_iterable()` instead of making a throw-away in-memory list.

Reply:
Without a list comprehension I think it would look like below which I find hard to read.
I went ahead with a third implementation that does not use a throw-away list either.

```python
return set(
    itertools.chain(
        builder.build_system_requires,
        itertools.chain.from_iterable(
            builder.get_requires_for_build(dist) for dist in distributions
        ),
    )
)
```

Reply:
Okay, marking this thread as resolved.